api/src/controllers/team.js

```diff
@@ -183,6 +172,78 @@ function updateTeamMember(req, res, next) {
     .catch((err) => next(fillErrorObject(500, 'Server error', [err])));
 }
 
+/**
+ * Delete the team member from the database on /team/:teamId/member/:member_id.
+ * @param {*} req request object, containing team id in the url
+ * @param {*} res response object, the relevant message returned
+ * @param next handler to the next middleware
+ * @returns 200: the team member was deleted
+ * @returns 404: team is not found
+ * @returns 400: team id is not in a valid hexadecimal format
+ */
+function deleteTeamMember(req, res, next) {
+  const { foundTeam } = req;
+  const { memberId } = req.params;
+  foundTeam.teamMembers.pull({ _id: memberId });
+  foundTeam
+    .save()
+    .then(() => res.status(200).json(foundTeam.teamMembers))
+    .catch((err) => next(fillErrorObject(500, 'Server error', [err])));
+}
+
+/**
+ * Associates a twitter handle with a team on the /team/twitter-handle/:teamId endpoint.
+ * @param {*} req request object, containing the teamId in the url and twitter handle in the body
+ * @param {*} res response object
+ * @param next handler to the next middleware
+ * @returns 200: successful added twitter handle to team
+ * @returns 400: team id is not in a valid hexadecimal format
+ * @returns 404: team is not found, or handle is invalid
+ * @returns 500: error trying to update the document in db
+ */
+async function storeHandle(req, res, next) {
+  const { twitterHandle: handle } = req.body;
+  const { foundTeam } = req;
+
+  if (handle.length === 0) {
+    // remove the handle from the doc
+    foundTeam.twitterHandle = '';
+  } else {
+    // update the handle
+    // validate the handle by getting user id
+    if (!process.env.TWITTER_BEARER_TOKEN) {
+      return next(
+        fillErrorObject(500, 'Missing environment variable', [
+          'No Twitter API Bearer Token found in .env file',
+        ]),
+      );
+    }
+    const response = await axios.get(
+      `https://api.twitter.com/2/users/by/username/${handle}`,
```

Comment:
Out of curiosity, Is octokit just for raw gitHub requests or can it be used here too?

Reply:
Octokit is the official GitHub client library. But good spot, we should upgrade this code to use the official Twitter library in the future.

I haven’t used it before in NodeJS but this seems to be the one https://www.npmjs.com/package/twitter-api-v2